ports/nrf/mphalport.c

```diff
@@ -35,6 +35,57 @@
 #include "nrfx_errors.h"
 #include "nrfx_config.h"
 
+#if MICROPY_PY_TIME_USE_RTC_BASE
+#include "nrfx_rtc.h"
+#endif
+
+#if MICROPY_PY_TIME_USE_TICKER_BASE
+#include "ticker.h"
+#endif
+
+
+#if MICROPY_PY_TIME_TICKS
+
+#if MICROPY_PY_TIME_USE_RTC_BASE
+nrfx_rtc_t rtc1 = NRFX_RTC_INSTANCE(1);
+
+// setup rtc1 for msec resolution, introduces about 0.7% error:
+// error = 0.001 / ( 1/32768 * (PRESCALER +1))
+// power consumption ~0.5uA
+void rtc1_init_msec(void) {
+    rtc1.p_reg->PRESCALER = 32;
+    rtc1.p_reg->TASKS_START = 1;
+}
+
+mp_uint_t mp_hal_ticks_ms(void) {
+    return (mp_uint_t)rtc1.p_reg->COUNTER;
```

Comment:
To make this more accurate, would it be possible to set the prescaler to 0, so the RTC counts at 32kHz, then return `COUNTER * 1000 / 32768`?  It would also need to keep track of roll-over of the RTC counter to get longer times, which could be done in an IRQ if the RTC has an IRQ?

Reply:
Good idea. 
It definetly helps with the long-term accuracy of the time, something that the user should except when running from the 32kHz crystal. It might not help for short time intervals (e.g. if you want to delay for 3msec). But for such short intervals, the user might need to use `time.usec` anyway.

> if the RTC has an IRQ?

yes it has one, and the overflow event can be set as irq.

I'll check this if I can come up with something.

Reply:
Sounds good.  It will actually have good ms-resolution, because the underling RTC counter has sub-ms resolution.